### PR TITLE
Add support for Power8

### PR DESCRIPTION
**Pull request by niroyb**
@Microsoft/ostc-devs

### CODE REVIEW

File: build/configure [hunk 1]
```diff
@@ -90,6 +90,11 @@ case $uname in
 	fi
 
         case `uname -m` in
+	    # ARCH is only used for OMS to determine the correct ruby folder
```
Comment (jeffaco) [inline build/configure]: Really? You sure on that? If so, that must have been a real joy to figure out! ☹️

File: source/code/include/scxcorelib/scxatomic.h [hunk 2]
```diff
@@ -113,6 +115,22 @@ __inline static bool scx_atomic_decrement_test(scx_atomic_t* v)
 #endif
 
 #elif defined(linux)
+
+// Use intrinsics for IBM powerPC
+#if defined(ppc)
+
+static __inline__ void scx_atomic_increment(scx_atomic_t* v)
+{
+    ++(*v);
```
Comment (jeffaco) [inline source/code/include/scxcorelib/scxatomic.h]: Um, is this REALLY atomic? Isn't there some sort of real "atomic increment" routine on PPC? We have atomic unit tests; do you know if our PPC platform VM was built with multiple CPUs for proper atomic tests?

File: source/code/scxsystemlib/common/scxsysteminfo.cpp [hunk 3]
```diff
@@ -406,6 +406,10 @@ namespace SCXSystemLib
             :          /* input operands                        */
                 "a" (static_cast< unsigned int >(function))
         );
+#elif defined(ppc)
+        // TODO add CPUID support for IBM PowerPC
+        (void) function;
+        (void) registers;
```
Comment (jeffaco) [inline source/code/scxsystemlib/common/scxsysteminfo.cpp]: I think this TODO item should be done prior to checkin of this code ... I'm going to mark this as another review needed due to this.

File: source/code/scxsystemlib/disk/staticdiskpartitionenumeration.cpp [hunk 4]
```diff
@@ -993,15 +993,14 @@ namespace SCXSystemLib
         
         try
         {
-            std::wstring command = parted_path + L" -l";
+            std::wstring command = parted_path + L" -ls";
             SCX_LOGTRACE(m_log, L"Invoking command : \"" + command + L"\"");
             int ret = m_deps->Run(command, processInput, processOutput, processErr, 15000);
 
-            std::string errOut = processErr.str();
```
Comment (jeffaco) [inline source/code/scxsystemlib/disk/staticdiskpartitionenumeration.cpp]: I think there was a prior bug here. Shouldn't processErr.str() have been dumped in trace as well?

File: source/code/util/base64/Base64Helper.cpp [hunk 5]
```diff
@@ -68,23 +68,23 @@ void Base64Helper::Encode(const std::vector<unsigned char>& input, std::string&
...
-        41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, -1, -1, -1, -1, -1,
-        -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1,
-        -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1,
-        -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1,
-        -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1,
-        -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1,
-        -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1,
-        -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1,
-        -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1,
+    static unsigned char base64ReverseTable[256] = {
```
Comment (jeffaco) [inline source/code/util/base64/Base64Helper.cpp]: Fun fun (not). I'll assume that this has built with PBUILD everywhere successfully. Be sure your .pbuild file is up to date for all of our platforms.

File: test/code/scxsystemlib/memory/memoryinstance_test.cpp [hunk 6]
```diff
@@ -695,11 +695,26 @@ class MemoryInstance_Test : public CPPUNIT_NS::TestFixture
...
+                */
+
+                keyValues["AvailableMemory"] = ToSCXUlong(topTokens[4 + toffset]);
+                keyValues["UsedMemory"]      = ToSCXUlong(topTokens[6 + toffset]);
+                keyValues["AvailableSwap"]   = ToSCXUlong(topTokens[14 + toffset]);
+                keyValues["UsedSwap"]        = ToSCXUlong(topTokens[16 + toffset]);
+#else
                 keyValues["AvailableMemory"] = ToSCXUlong(topTokens[6 + toffset]);
                 keyValues["UsedMemory"]      = ToSCXUlong(topTokens[4 + toffset]);
-                keyValues["TotalSwap"]       = ToSCXUlong(topTokens[12 + toffset]);
```
Comment (jeffaco) [inline test/code/scxsystemlib/memory/memoryinstance_test.cpp]: What happened to TotalSwap, and why?

File: source/code/include/scxcorelib/scxatomic.h (hunk 2 above)
Comment (niroyb) [inline source/code/include/scxcorelib/scxatomic.h]: This is using [c++ 11 atomics ](http://en.cppreference.com/w/cpp/atomic/atomic/operator_arith)
We do get a warning on osdevppc-rhel71-05 that the test should be run on a multiple CPU machine although `lscpu` shows 32 CPUs. Will investigate some more.

File: test/code/scxsystemlib/memory/memoryinstance_test.cpp (hunk 6 above)
Comment (niroyb) [inline test/code/scxsystemlib/memory/memoryinstance_test.cpp]: It's at line 698. Only the available and swap memory token locations differ in the output.

Comment (jeffaco) [inline test/code/scxsystemlib/memory/memoryinstance_test.cpp]: I get it now. This is rather spaghetti-ish. Please "unmingle" to be more clear with appropriate comments in each section as per our discussion.

File: source/code/scxsystemlib/common/scxsysteminfo.cpp (hunk 3 above)
Comment (jeffaco) [inline source/code/scxsystemlib/common/scxsysteminfo.cpp]: Please make other changes as we discussed and push again ...

File: test/code/scxsystemlib/memory/memoryinstance_test.cpp [hunk 7]
```diff
@@ -695,9 +710,9 @@ class MemoryInstance_Test : public CPPUNIT_NS::TestFixture
                 }
 
                 keyValues["TotalMemory"]     = ToSCXUlong(topTokens[2 + toffset]);
+                keyValues["TotalSwap"]       = ToSCXUlong(topTokens[12 + toffset]);
```
Comment (jeffaco) [inline test/code/scxsystemlib/memory/memoryinstance_test.cpp]: Please change ordering to eliminate needless code churn. This needs to be town lines down, it looks like.

**Pull request merged**